python/tests/kernel/test_kernel_features.py

```diff
@@ -344,3 +342,68 @@ def test(q: int, p: int):
     counts = cudaq.sample(test, 2, 0)
     counts.dump()
     assert '01100' in counts and len(counts) == 1
+
+
+def test_no_dynamic_lists():
+    with pytest.raises(RuntimeError) as error:
+        @cudaq.kernel(jit=True)
+        def kernel(params : list[float]):
+            params.append(1.0)
+
+    with pytest.raises(RuntimeError) as error:
+        @cudaq.kernel(jit=True)
+        def kernel():
+            l = [i for i in range(10)]
+            l.append(11)
+        print(kernel)
+    
+    with pytest.raises(RuntimeError) as error:
+        @cudaq.kernel(jit=True)
+        def kernel():
+            l = [[i,i,i] for i in range(10)]
+            l.append([11,12,13])
+        print(kernel)
+
+def test_simple_return_types():
+    @cudaq.kernel
+    def kernel(a:int, b:int) -> int:
+        return a * b
+
+    ret = kernel(2, 4)
+    assert ret == 8
+
+    @cudaq.kernel
+    def qernel(a:float, b:float) -> float:
+        return a * b
+
+    ret = kernel(2, 4)
+    assert np.isclose(ret, 8., atol=1e-12)
+
+    with pytest.raises(RuntimeError) as error:
+        @cudaq.kernel
+        def kernel(a:int, b:int): # No return type
+            return a * b
+
+
+def test_list_creation():
+
+    N = 10 
+    @cudaq.kernel
+    def kernel(N :int, idx: int) -> int:
+        myList = [i+1 for i in range(N-1)]
+        return myList[idx]
+
```

Comment:
Q: Is this allowed?
`qubits_list = list(range(N))`
where, `N:int` is kernel argument.

Reply:
`list(...)` is not yet implemented, but I can add that here.